src/python/pants/backend/scala/subsystems/scalac.py

```diff
@@ -62,7 +62,7 @@ class Scalac(Subsystem):
         )
         .advanced()
         .deprecated(
-            removal_version="2.12.0dev0",
+            removal_version="2.12.0.dev0",
```

Comment:
Oops. Looks like maybe we lost eager validation that these version strings are valid? Not sure if related to the new options system. cc @thejcannon

Reply:
I'd be moderately surprised if that were true. The new options system acts as a passthrough, so if you're seeing the removal info when invoking `./pants help ...` then the passthrough is working (and the rest of the machinery is firing like before)